shared/reducers/unlock-folders.js

```diff
@@ -0,0 +1,76 @@
+/* @flow */
+
+import * as Constants from '../constants/unlock-folders'
+import HiddenString from '../util/hidden-string'
+
+import type {UnlockFolderActions, Device} from '../constants/unlock-folders'
+
+export type State = {
+  phase: 'dead' | 'promptOtherDevice' | 'paperKeyInput' | 'success',
+  devices: ?Array<Device>,
+  paperkey: ?HiddenString,
```

Comment:
Ideally we wouldn't use this, so whoever does the functional thing should avoid it